qa/packaging/src/test/java/org/elasticsearch/packaging/test/DockerTests.java

```diff
@@ -1220,6 +1219,7 @@ public void test500Readiness() throws Exception {
             builder().envVar("readiness.port", "9399").envVar("xpack.security.enabled", "false").envVar("discovery.type", "single-node")
         );
         waitForElasticsearch(installation);
+        dumpDebug();
         assertTrue(readinessProbe(9399));
```

Comment:
Could we have some form of `assertTrue` that takes a `Callable` that's executed when the condition evaluates to `false`? That we we can just keep this debug bit in here.

Reply:
I think that will be just as flaky as the current situtation (since we still might have various asserts). What I'd really like is to have any failure in PackagingTestCase call dumpDebug. We already handle and edge case of rolling over logs, where we already know if the test failed or not. But we have dumpDebug now scattered in the test infra, and I don't want to duplicate it.

While I do think this can be improved, here I was going for a quick and isolated addition of debugging to diagnose this specific case.

Reply:
> I think that will be just as flaky as the current situtation (since we still might have various asserts). What I'd really like is to have any failure in PackagingTestCase call dumpDebug.

Yeah, that's kind of what I was getting at. A general solution would be great. We can definitely address that in a follow up though.